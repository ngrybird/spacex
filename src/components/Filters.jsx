import React from "react";
import { FILTER_YEARS } from "../utils/constants";

class Filters extends React.Component {
  constructor(props) {
    super(props);
    this.state = { ...this.props };
  }
  onYearClick = e => {
    this.setState(
      { ...this.state, year: e.target.value },
      this.updateFiltersData
    );
  };
  toggleLanding = e => {
    this.setState(
      { ...this.state, successLand: e.target.value },
      this.updateFiltersData
    );
  };
  toggleLaunch = e => {
    this.setState(
      { ...this.state, successLaunch: e.target.value },
      this.updateFiltersData
    );
  };
  updateFiltersData = () => {
    this.props.updateFilters(this.state);
  };
  render() {
    return (
      <div className="filters-container">
        <b>Filters :</b>

        <div className="filters-content">
          <div className="labels">
            <u>Launch Year</u>
          </div>
          {FILTER_YEARS.map(year => (
            <button
              className={this.props.year === year ? "is-selected" : ""}
              onClick={this.onYearClick}
              key={year}
              value={year}
            >
              {year}
            </button>
          ))}
        </div>
        <div className="filters-content">
          <div className="labels">
            <u>Successful Launch</u>
          </div>
          <div>
            <button
              onClick={this.toggleLaunch}
              value={true}
              className={
                this.props.successLaunch === "true" ? "is-selected" : ""
              }
            >
              True
            </button>
            <button
              onClick={this.toggleLaunch}
              value={false}
              className={
                this.props.successLaunch === "false" ? "is-selected" : ""
              }
            >
              False
            </button>
          </div>
        </div>
        <div className="filters-content">
          <div className="labels">
            <u>Successful Landing</u>
          </div>
          <div>
            <button
              onClick={this.toggleLanding}
              value={true}
              className={this.props.successLand === "true" ? "is-selected" : ""}
            >
              True
            </button>
            <button
              onClick={this.toggleLanding}
              value={false}
              className={
                this.props.successLand === "false" ? "is-selected" : ""
              }
            >
              False
            </button>
          </div>
        </div>
      </div>
    );
  }
}
export default Filters;
