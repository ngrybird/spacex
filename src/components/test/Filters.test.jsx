import { render, fireEvent } from "@testing-library/react";
import React from "react";
import { apiResponse } from "./testData";
import Filters from "../Filters";

describe("<Filters/>", () => {
  const props = {
    filterOptions: {
      year: "",
      successLaunch: "",
      successLand: ""
    },
    updateFilters: jest.fn()
  };
  it("render filters component", () => {
    const { container } = render(<Filters {...props} />);
    expect(container).toMatchSnapshot();
  });
  it("simulate click on year button", () => {
    const { container, getByText } = render(<Filters {...props} />);
    fireEvent.click(getByText("2019"));
    expect(props.updateFilters).toHaveBeenCalled();
  });
  it("simulate click on True button for Successful launch", () => {
    const { container, getAllByText } = render(<Filters {...props} />);
    fireEvent.click(getAllByText("True")[0]);
    expect(props.updateFilters).toHaveBeenCalled();
  });
  it("simulate click on True button for Successful landing", () => {
    const { container, getAllByText } = render(<Filters {...props} />);
    fireEvent.click(getAllByText("True")[1]);
    expect(props.updateFilters).toHaveBeenCalled();
  });
});
