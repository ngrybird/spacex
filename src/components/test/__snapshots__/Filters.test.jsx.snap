// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters/> render filters component 1`] = `
<div>
  <div
    class="filters-container"
  >
    <b>
      Filters :
    </b>
    <div
      class="filters-content"
    >
      <div
        class="labels"
      >
        <u>
          Launch Year
        </u>
      </div>
      <button
        class=""
        value="2006"
      >
        2006
      </button>
      <button
        class=""
        value="2007"
      >
        2007
      </button>
      <button
        class=""
        value="2008"
      >
        2008
      </button>
      <button
        class=""
        value="2009"
      >
        2009
      </button>
      <button
        class=""
        value="2010"
      >
        2010
      </button>
      <button
        class=""
        value="2011"
      >
        2011
      </button>
      <button
        class=""
        value="2012"
      >
        2012
      </button>
      <button
        class=""
        value="2013"
      >
        2013
      </button>
      <button
        class=""
        value="2014"
      >
        2014
      </button>
      <button
        class=""
        value="2015"
      >
        2015
      </button>
      <button
        class=""
        value="2016"
      >
        2016
      </button>
      <button
        class=""
        value="2017"
      >
        2017
      </button>
      <button
        class=""
        value="2018"
      >
        2018
      </button>
      <button
        class=""
        value="2019"
      >
        2019
      </button>
      <button
        class=""
        value="2020"
      >
        2020
      </button>
    </div>
    <div
      class="filters-content"
    >
      <div
        class="labels"
      >
        <u>
          Successful Launch
        </u>
      </div>
      <div>
        <button
          class=""
          value="true"
        >
          True
        </button>
        <button
          class=""
          value="false"
        >
          False
        </button>
      </div>
    </div>
    <div
      class="filters-content"
    >
      <div
        class="labels"
      >
        <u>
          Successful Landing
        </u>
      </div>
      <div>
        <button
          class=""
          value="true"
        >
          True
        </button>
        <button
          class=""
          value="false"
        >
          False
        </button>
      </div>
    </div>
  </div>
</div>
`;
